src/V3MergeCond.cpp

```diff
@@ -337,7 +319,7 @@ class MergeCondVisitor final : public VNVisitor {
         m_mgCondp = nullptr;
         m_mgLastp = nullptr;
         m_mgNextp = nullptr;
-        m_markVars.clear();
+        AstNode::user1ClearTree();  // Clear marked variables
         AstNode::user2ClearTree();
```

Comment:
Unrelated, for a future cleanup I think AstNode::user1ClearTree should instead be a clear operation executed on the user1inuse object. Then it's clear that the code has "permission" to do the clear.   (The user1ClearTree function far predates tho